src/components/lobby/create_game.tsx

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable operator-linebreak */
+import React, { useState } from 'react';
+import { WebSocketConnection } from '../../types';
+
+import GameTypes from '../../modules/game/gameTypes';
+import OutgoingAction from '../../modules/ws/outgoing_action';
+import MahjongVersions from '../../modules/mahjong/versions';
+import BigTwoVersions from '../../modules/bigtwo/versions';
+
+type CreateGameFormProps = {
+  ws?: WebSocketConnection | null;
+};
+
+/**
+ * Form component used to send a request to WebSocket API to create a game.
+ */
+const CreateGameForm = ({ ws }: CreateGameFormProps): JSX.Element => {
+  /**
+   * States.
+   */
+  const [gameName, setGameName] = useState<string>('');
+  const [gameType, setGameType] = useState<string>(GameTypes.Mahjong);
+  const [gameVersion, setGameVersion] = useState<string>(MahjongVersions.HongKong);
+
+  /**
+   * State Handlers.
+   */
+  const handleSetGameName = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setGameName(event.target.value);
+  };
+
+  const handleSetGameType = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const specifiedGameType = event.target.value as string;
+    setGameType(specifiedGameType);
+    if (specifiedGameType === GameTypes.Mahjong) {
+      setGameVersion(MahjongVersions.HongKong);
+    }
+    if (specifiedGameType === GameTypes.BigTwo) {
+      setGameVersion(BigTwoVersions.Chinese);
+    }
+  };
+
+  const handleSetGameVersion = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setGameVersion(event.target.value);
+  };
+
+  /**
+   * Create Game Function
+   */
+  const createGame = (event: React.FormEvent<HTMLInputElement>): void => {
+    event.preventDefault();
+
+    const payload = {
+      game: {
+        gameName,
+        gameType,
+        gameVersion,
+      },
+    };
```

Comment:
can make helper functions to construct the payload

Reply:
Done